### PR TITLE
P2022-1327/MainActivity fragment navigation.

### PR DESCRIPTION
**Pull request by Lucas-Fresko**
Adds 3 fragments to MainActivity and navigation between them using tablayout.

Moves temporary navigation to Details Activity to Home fragment.

### CODE REVIEW

File: app/src/main/java/com/intive/patronage22/intivi/Adapter/MainFragmentsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+internal class MainFragmentsAdapter (fa: FragmentActivity, var totalTabs: Int): FragmentStateAdapter(fa) {
+
+    enum class ActiveTab {
+        HOME, FAVOURITES, GENRES
+    }
+
+    override fun createFragment(position: Int): Fragment {
+        return when(position){
+            ActiveTab.HOME.ordinal -> {
+                Log.d("bayraktar", "Adapter calls home")
```
Comment (taurelas) [inline app/src/main/java/com/intive/patronage22/intivi/Adapter/MainFragmentsAdapter.kt:18]: please remove the log messages, we don't want to be targeted by a drone :)

File: app/src/main/java/com/intive/patronage22/intivi/Adapter/MainFragmentsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                HomeFragment()
+            }
+            ActiveTab.FAVOURITES.ordinal-> {
+                Log.d("bayraktar", "Adapter calls favourites")
```
Comment (taurelas) [inline app/src/main/java/com/intive/patronage22/intivi/Adapter/MainFragmentsAdapter.kt:22]: log to be removed

**Review by taurelas — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/intive/patronage22/intivi/MainActivity.kt [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
 
-    private fun clickNavigate(view: View, activity: Class<*>){
-        view.setOnClickListener{
-            startActivity(Intent(this, activity))
-        }
+        tabLayout = findViewById(R.id.footer_tab)
+        viewPager = findViewById(R.id.main_viewPager)
+        Log.d("bayraktar", "wtf is this")
+
+        val adapter = MainFragmentsAdapter(this, tabLayout.tabCount)
```
[the remark this answers is not in the document]
Comment (Lucas-Fresko) [reply app/src/main/java/com/intive/patronage22/intivi/MainActivity.kt:24]: I did that before, but it complained with some error so I left it... strangely it's not complaining this time. Code is such a fickle thing.

File: app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".Fragments.HomeFragment"
+    android:id="@+id/frameLayout">
+
+    <LinearLayout
+        android:id="@+id/header_bar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="12dp"
```
[the remark this answers is not in the document]
Comment (Lucas-Fresko) [reply app/src/main/res/layout/fragment_home.xml:14]: We have decided to stop putting everything into dimens but we can discuss this again during the next meeting.

File: app/src/main/java/com/intive/patronage22/intivi/Fragments/FavouritesFragment.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.intive.patronage22.intivi.R
+
+class FavouritesFragment : Fragment() {
+
+
```
Comment (kosiara) [inline app/src/main/java/com/intive/patronage22/intivi/Fragments/FavouritesFragment.kt:12]: avoid double empty lines

**Pull request merged**